verticapy/tests_new/core/vdataframe/test_math.py

```diff
@@ -0,0 +1,885 @@
+                    ["Beyonce", "Alicia Keys", "Dr Dre"],
+                ],
+                "album_cost": [
+                    [65, 50, 90.11, 25, 71],
+                    [40, 50, 90.11, 35],
+                    [56, 50, 90.11, 55, 213],
+                ],
+                "sign_num": [0, -1, 2],
+            }
+        ).to_vdf()
+        vpy_data = sample_data if data == "sample_data" else titanic_vd_fun
+
+        sample_data_pdf = sample_data.to_pandas()
+        py_data = sample_data_pdf if data == "sample_data" else titanic_pdf
+        py_data[columns] = (
+            py_data[columns].astype(float)
+            if vpy_data[columns].isnum() and not vpy_data[columns].isarray()
+            else py_data[columns]
+        )
+
+        vpy_data[columns].apply_fun(func=vpy_func)
+        vpy_res = vpy_data[columns].sum()
+
+        py_data[columns] = py_data[columns].apply(lambda x: eval(py_func))
+        py_res = float(py_data[columns].sum())
+
+        print(
+            f"Function Name: {vpy_func}, \nVerticaPy Result: {vpy_res} \nPython Result :{py_res}\n"
+        )
+        assert vpy_res == pytest.approx(py_res)
+
+    @pytest.mark.parametrize(
+        "part, columns",
+        [
+            ("hour", "time"),
+            ("minute", "time"),
+            ("second", "time"),
+            ("microsecond", "time"),
+            ("day", "time"),
+            ("month", "time"),
+            ("year", "time"),
+            ("quarter", "time"),
+        ],
+    )
+    def test_date_part(self, smart_meters_vd, part, columns):
+        """
+        test function - date_part
+        """
+        smart_meters_copy = smart_meters_vd.copy()
+        smart_meters_pdf = smart_meters_vd.to_pandas()
+
+        vpy_res = smart_meters_copy[columns].date_part(part)[columns].sum()
+
+        py_res = getattr(smart_meters_pdf[columns].dt, part).sum()
+
+        print(
+            f"Date Part: {part} \nVerticaPy Result: {vpy_res} \nPython Result :{py_res}\n"
+        )
+
+        assert vpy_res == pytest.approx(py_res)
+
+    @pytest.mark.parametrize("col_type", (["complex", "string"]))
+    def test_get_len(self, titanic_vd_fun, laliga_vd, col_type):
+        """
+        test function - get_len
+        """
+        titanic_pdf = titanic_vd_fun.to_pandas()
+        laliga_pdf = laliga_vd.to_pandas()
+
+        if col_type == "complex":
+            vpy_res = laliga_vd["away_team"]["managers"][0]["name"].get_len().sum()
+            py_res = (
+                laliga_pdf["away_team"]
+                .apply(lambda x: len(x["managers"][0]["name"]) if x["managers"] else 0)
+                .sum()
+            )
+        else:
+            vpy_res = titanic_vd_fun["name"].get_len().sum()
+            py_res = titanic_pdf["name"].apply(len).sum()
+
+        print(f"VerticaPy Result: {vpy_res} \nPython Result :{py_res}\n")
+
+        assert vpy_res == pytest.approx(py_res)
+
+    @pytest.mark.parametrize("column, n", ([("age", 4), ("fare", 2)]))
```

Comment:
age in titanic is itneger and not float so it may not be the ideal column.

again i would suggest just using a small dummy dataset. 

vdf= vp.vDataFrame({"col1":[-1, 1.22, 2], "col2":[-10.33, -2.0, 3]})

Maybe create one dataset for this entire exercise but keep it very simple. And broad such that it covers everything

Reply:
I have used age and fare columns. Both are having float values in dataset.
for consistency, I am using titanic dataset as its being used by other test functions in this file.

Agree, we can use small dataset for this test.